src/dialog-editor/services/dialogValidationService.ts

```diff
@@ -0,0 +1,66 @@
+import {__} from '../../common/translateFunction';
+import * as _ from 'lodash';
+
+export default class DialogValidationService {
+  public invalid: any = {};
+
+  /** @ngInject */
+  constructor(private DialogEditor: any) {
```

Comment:
Is this a good idea? Is this even possible?

Reply:
@skateman I need the data from the `DialogEditor` service. Tested and works fine. Any idea would you like to improve it?

Reply:
Is this REALLY a service if it provides data? I'm not THAT familiar with angular but AFAIK services are singletons that should be used/called in controllers and the data should be passed as an argument and not as a dependency. But maybe I'm wrong so pinging @himdel @karelhala

Reply:
Probably too late for such a design change now. But.. using services as a data/state store is quite common in angular world.

(I guess it's not yet the time to move this to redux, right?)

Reply:
Ah.. but it's not too late not to rely on DialogEditor here, we can just pass the data as an argument to `dialogIsValid` - cleaner that way probably :)

Reply:
Forgot to add my comment before. To store current data in service is OK to do, however while we are thinking to use redux, we should reimplement the `dialogIsValid` function to accept dialog data. That way once we move this to use redux we can control them by redux and they will be passed to this function every time they change.